presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -495,6 +504,19 @@ private static boolean shouldUseFileSplitsFromInputFormat(InputFormat<?, ?> inpu
         return splitList;
     }
 
+    private List<InternalHiveSplit> getVirtuallyBucketedSplits(Path path, FileSystem fileSystem, InternalHiveSplitFactory splitFactory, int bucketCount)
+    {
+        // List all files recursively in the partition and assign virtual bucket number to each of them
+        return stream(new HiveFileIterator(path, fileSystem, directoryLister, namenodeStats, RECURSE))
+                .map(file -> {
+                    int virtualBucketNumber = getVirtualBucket(bucketCount, file.getPath());
+                    return splitFactory.createInternalHiveSplit(file, OptionalInt.of(virtualBucketNumber), OptionalInt.empty(), false);
```

Comment:
nit: why make the split not splittable? cc @jessesleeping

Reply:
per offline discussion, have "read bucket" present but "table bucket" not present is quite difficult to understand. Two thoughts:

- Add specific "virtual bucket number". When this is presented, other "xxx bucket" shouldn't be present.
- Set "table bucket" and "read bucket" to same.

Reply:
@wenleix 
`splittable` was introduced after I wrote the commit. I think it's fine to pass down the `splittable` flag to virtual bucketed internal split.